review/jm-initial

fivetran-joemarkiewicz opened the pull request:
This PR is not intended to be merged, but rather acts as an initial review of the first build of the package. Please see the comments below for any notes to apply to the `main` branch prior to the initial release.

fivetran-joemarkiewicz commented on models/stg_zuora.yml:
```diff
@@ -74,9 +74,9 @@ models:
-        tests:
-          - accepted_values:
-              values: ['Assets', 'Liabilities', 'Equity', 'Revenue', 'Expenses']
+        tests: 
+          - accepted_values: 
+              values: ['Assets', 'Liabilities', 'Equity', 'Revenue', 'Expenses'] 
```
These types of tests have typically come back to burn us in the long run since the source can change without us being prepared for the change. Unless it is paramount that the field have **only** the accepted values, then I would request we remove these tests.

I don't want to over test and have users with production environments be presented with a number of test failures. Would you be able to remove all the accepted_values tests in this package.

fivetran-avinash replied:
Removed.  Good point about changes in the source.

fivetran-joemarkiewicz replied:
It looks like one is still left in the tests. Can we remove it [here](https://github.com/fivetran/dbt_zuora_source/blob/dec2f5f2553787b54b548165b3e38d6f4b4bf192/models/stg_zuora.yml#L146-L148)

fivetran-joemarkiewicz reviewed (commented):
@fivetran-avinash thanks for working through the initial build of the source package! I have a few comments and call outs in this review. Would you be able to review my comments and make the appropriate updates in `main` and respond in the threads once the updates are applied. I can then resolve the threads during the re-review.

In addition, I wanted to call out that I think we are bringing in too many fields for a number of the staging models. I don't feel there is much needed to be done about this now, but do want to make sure we come back to this following the end models build out so we can trim some more of the fields that are not necessary. Also, we can finish the yml documentation at that moment as well.

fivetran-avinash commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -1,7 +1,7 @@
-  - name: Ask a question during our office hours
-    url: https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours
-    about: Schedule time during the external office hours block with the Fivetran Analytics Engineering team for support
+  - name: Ask a question during our office hours 
+    url: https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours 
+    about: Schedule time during the external office hours block with the Fivetran Analytics Engineering team for support 
```
oh interesting I copied/pasted from another package. Did we bulk update this across packages?

fivetran-joemarkiewicz replied:
We are doing a bit of A/B testing to see the impact and benefit of continue to point folks to office hours vs pointing to the community. For new packages we should point to the community. I imagine we will want to scale out to the res of the packages at some point.

I don't have any copy for this specific community point, it would be great if you could come up with one and we can review and look to leverage in other package.

fivetran-joemarkiewicz replied:
This is great, thanks @fivetran-avinash!

fivetran-avinash commented on .buildkite/hooks/pre-command:
```diff
@@ -19,6 +19,6 @@ export CI_SNOWFLAKE_DBT_PASS=$(gcloud secrets versions access latest --secret="C
 export CI_SNOWFLAKE_DBT_USER=$(gcloud secrets versions access latest --secret="CI_SNOWFLAKE_DBT_USER" --project="dbt-package-testing-363917")
 export CI_SNOWFLAKE_DBT_WAREHOUSE=$(gcloud secrets versions access latest --secret="CI_SNOWFLAKE_DBT_WAREHOUSE" --project="dbt-package-testing-363917")
-export CI_DATABRICKS_DBT_HOST=$(gcloud secrets versions access latest --secret="CI_DATABRICKS_DBT_HOST" --project="dbt-package-testing-363917")
+export CI_DATABRICKS_DBT_HOST=$(gcloud secrets versions access latest --secret="CI_DATABRICKS_DBT_HOST" --project="dbt-package-testing-363917") 
```
added!

fivetran-avinash commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -1,7 +1,7 @@
 description: Report a bug or an issue you've found within the dbt package
 title: "[Bug] <title>"
-labels: ["bug", "triage"]
+labels: ["bug", "triage"] 
```
Uhh what are the current versions? I'm seeing `labels: ["bug", "triage"]` across multiple packages.

fivetran-avinash commented on .github/ISSUE_TEMPLATE/feature-request.yml:
```diff
@@ -1,7 +1,7 @@
 name: 🎉 Feature
 description: Suggest a new feature for the Fivetran dbt package
 title: "[Feature] <title>"
-labels: ["enhancement"]
+labels: ["enhancement"] 
```
Same comment as above, but updated to `labels: ["enhancement","update_type:feature"] `

fivetran-joemarkiewicz replied:
Let's keep it as just `type:enhancement`. Let's let the update_type be up to us when we are triaging the ticket.

fivetran-avinash replied:
Sounds good, updated!

fivetran-avinash commented on .gitignore:
```diff
@@ -3,4 +3,4 @@ target/
 .DS_Store
-dbt_packages/
\ No newline at end of file
+dbt_packages/ 
```
Added!

fivetran-avinash commented on dbt_project.yml:
```diff
@@ -29,10 +29,10 @@ vars:
-  zuora_rate_plan_pass_through_columns: []
-  zuora_rate_plan_charge_pass_through_columns: []
-  zuora_subscription_pass_through_columns: []
+  zuora_account_pass_through_columns: [] 
+  zuora_rate_plan_pass_through_columns: [] 
+  zuora_rate_plan_charge_pass_through_columns: [] 
+  zuora_subscription_pass_through_columns: [] 
```
Updated.

When querying development data on `account` and `subscription `I encountered fields that appeared to be custom. Example: `select * from `singular-vector-135519.zuora_sandbox_v2.subscription` yields a `custom_date_field_c`.

Removed rate_plan and rate_plan_charge--I thought I found custom fields the first time but that must have been a misread.

fivetran-joemarkiewicz replied:
Thanks for the clarification, this logic makes sense to me for account and subscription! Good catch on adjusting the other models.

fivetran-avinash commented on integration_tests/.gitignore:
```diff
@@ -1,4 +1,4 @@
 dbt_modules/
-logs/
\ No newline at end of file
+logs/ 
```
Added!

fivetran-avinash commented on integration_tests/ci/sample.profiles.yml:
```diff
@@ -45,7 +45,7 @@ integration_tests:
       threads: 8
     databricks:
-      catalog: null
+      catalog: null 
```
Updated.

fivetran-avinash commented on macros/get_amendment_columns.sql:
```diff
@@ -1,8 +1,8 @@
-There are 1 unused configuration paths:
-- models.development.example
+[0m01:16:44  [[33mWARNING[0m]: Configuration paths exist in your dbt_project.yml file which do not apply to any resources. 
+There are 1 unused configuration paths: 
+- models.development.example 
 
-[0m01:16:48  {% macro get_amendment_columns() %}
+[0m01:16:48  {% macro get_amendment_columns() %} 
```
Fixed.

fivetran-avinash commented on macros/get_subscription_columns.sql:
```diff
@@ -1,4 +1,4 @@
- {% macro get_subscription_columns() %}
+{% macro get_subscription_columns() %}
```
Updated.

fivetran-avinash commented on models/stg_zuora__account.sql:
```diff
@@ -51,7 +51,9 @@ final as (
         updated_by_id,
         updated_date,
         vatid
+
         {{ fivetran_utils.fill_pass_through_columns('zuora_account_pass_through_columns') }}
+        
```
Updated!

fivetran-avinash commented on models/src_zuora.yml:
```diff
@@ -7,9 +7,9 @@ sources:
-    freshness:
-      warn_after: { count: 72, period: hour }
-      error_after: { count: 168, period: hour }
+    freshness: 
+      warn_after: { count: 72, period: hour } 
+      error_after: { count: 168, period: hour } 
```
Good question. I replicated the Zuora freshness tests from Recurly since the use case was similar (subscription models usually will not need daily updates so we can extend it out to biweekly/weekly checks). I think for now global makes sense unless we think item tables need to be checked more frequently?

fivetran-avinash commented on integration_tests/dbt_project.yml:
```diff
@@ -32,4 +32,5 @@ vars:
-    search_order: ['spark_utils', 'dbt_utils']
\ No newline at end of file
+    search_order: ['spark_utils', 'dbt_utils']
+
```
Good question. Bulidkite threw no errors when I was testing, and it seemed like everything ran?  🤔  

I've added the seed configurations now and modified he files and everything passes!

fivetran-joemarkiewicz replied:
Great! Yeah if no seed config is needed then that is fine as well, but adding the config is probably best for future proofing the buildkite deployment. Definitely curious though how it worked right out of the box!

fivetran-avinash reviewed (commented):
Hi @fivetran-joemarkiewicz ! Addressed the PR as best I could, had some questions so feel free to check them. 

I'm unsure why the seed configuration passed though, but I've updated it. Just one follow-up question: for seed configurations, do we have a default setup for different fields, and should we apply it to all fields of those types?

What I did for this `dbt_project.yml` type, based on what I saw from other packages: 

float type: `"{{ 'FLOAT64' if target.type == 'bigquery' else 'float' }}"`
int type "`{{ 'INT64' if target.type == 'bigquery' else 'bigint' }}"`
timestamp type  `"timestamp"`

fivetran-joemarkiewicz commented:
Thanks for addressing the review notes @fivetran-avinash! I was able to resolve most and responded to the remaining ones.
Let me know if you have any final questions when addressing the final notes.

Additionally, in regards to the seed data configs the defaults are pretty consistent with what you provided. However, there can sometimes be cases where we need to adjust a bit more based on the target.type. Although those are usually case by case.

fivetran-avinash reviewed (commented):
@fivetran-joemarkiewicz Re-review changes applied!

And yeah addressing the excessive number of fields--agreed, and I'm pretty sure more consolidation is coming as I work through the packages.

But without much customer feedback, I decided to just err on the cautious side and bring more fields in that might be relevant, since it's easier to remove data in later stages of development rather than realizing later on that something is needed and we need to bring fields in later on (particularly with seed file creation).  I think either approach has drawbacks (mine definitely takes longer in earlier stages), but trying this for now and we'll see if it pays off long-term!

fivetran-joemarkiewicz reviewed (commented):
@fivetran-avinash thanks for working through those updates! This looks good to go! The only last request is to remove the `.DS_Store` file from the directory. This isn't needed and is a result of generating the docs prior to the `.gitignore` removing the file from the commit.
